Develop

BirmacherAkos opened the pull request:
Direct upload

BirmacherAkos committed:
Workflow clean + log clean (+7 squashed commits)
Squashed commits:
[a209375] comment
[09dfaf8]  -
[a7587ec] Android, ios
[1627388]  -
[5b93fc8] desc
[5a0f4b5] clean (+1 squashed commit)
Squashed commits:
[218bc44] Direct upload
[af33193] develop branch init

trapacska commented on appetize/client.go:
```diff
@@ -0,0 +1,173 @@
+type Client struct {
+	token     string
+	baseURL   string
+	client    *http.Client
```
```
client := Client{}
client.client <- warning!!! ⚠️
```

trapacska commented on main.go:
```diff
@@ -0,0 +1,139 @@
+	log.Infof("Checking provided file extension")
+	var pth string
+	var err error
+	if pth, err = ensureExtension(cfg.AppPath, cfg.Platform); err != nil {
```
the 2 lines above is unneccessary if you remove the `if` here

trapacska commented on main.go:
```diff
@@ -0,0 +1,139 @@
+
+	fmt.Println()
+	log.Infof("Upload")
+	// curl https://tok_vjbxr9m95cwe7r3fpjpqh3y94w@api.appetize.io/v1/apps -F "file=@/Users/birmachera/Desktop/ip/XcodeArchiveTest.ipa" -F "platform=ios"
```
pls don't 😄

trapacska commented on main.go:
```diff
@@ -0,0 +1,139 @@
+}
+
+func logDebugPretty(v interface{}) {
+	b, err := json.MarshalIndent(v, "", "  ")
```
this shall not even run if debug mode is off. please check debug mode here as well and skip marshalling as well

trapacska commented on main.go:
```diff
@@ -0,0 +1,139 @@
+		log.Printf("Updating the provided app at Appetize.io")
+	}
+
+	var response appetize.Response
```
these lines are not required if you don't use if like in the line below

trapacska commented on main.go:
```diff
@@ -0,0 +1,139 @@
+// For iOS platform the accepted extensions: .app, .zip, .tar.gz.
+// For Android platform the accepted extension: .apk.
+// If the platorm is iOS and the given file's extension is .app it creates a new .zip and returns the .zip's path.
+func ensureExtension(pth, platform string) (string, error) {
```
we can, so let decide the platform by ourselves so this block of code won't be required 👍

trapacska commented on step.yml:
```diff
@@ -0,0 +1,84 @@
+support_url: https://github.com/bitrise-steplib/steps-appetize-io-deploy/issues
+
+project_type_tags:
+  - ios
```
if all of our platforms can use this, please leave this list empty rather than list all of them

trapacska commented on step.yml:
```diff
@@ -0,0 +1,84 @@
+  - app_path:
+    opts:
+      title: Application path
+      summary: The full path to the application file (.app or .apk)
```
I would list all of the possibilities. .zip and .gz as well. Also I would split into cases like for android: apk, for ios: app, zip, ... also in desc.

trapacska commented on step.yml:
```diff
@@ -0,0 +1,84 @@
+      description: The full path to the application file (.app or .apk)
+      is_expand: true
+      is_required: true
+  - platform: "ios"
```
don't forget to remove

trapacska commented on step.yml:
```diff
@@ -0,0 +1,84 @@
+      is_required: false
+  - verbose: false
+    opts:
+      category: Debug
```
please do not use category for one item

trapacska commented on step.yml:
```diff
@@ -0,0 +1,84 @@
+  - verbose: false
+    opts:
+      category: Debug
+      title: Enable debug logging
```
please use the verbose word instead of debug

trapacska reviewed (changes requested):
please see my comments, otherwise 👍 

also: please keep in mind that I will need to review fully again after the changes. I will test the logic that time.

BirmacherAkos committed:
Comments (+1 squashed commit)
Squashed commits:
[7d9090b] step.yml clean (+1 squashed commit)
Squashed commits:
[b5d7823] Clean (+1 squashed commit)
Squashed commits:
[ad0850c] Artifact type witch extension, platform enums. (+1 squashed commit)
Squashed commits:
[6a49ccf] PR clean - fix

trapacska commented on appetize/until.go:
```diff
@@ -0,0 +1,140 @@
+	extension := path.Ext(filePath)
+
+	switch extension {
+	case ".zip":
```
https://github.com/golang/go/wiki/Switch#multiple-cases

please merge them 🙃

trapacska commented on appetize/until.go:
```diff
@@ -0,0 +1,140 @@
+
+	switch extension {
+	case ".zip":
+		art := Artifact{filePath: filePath, platform: IOS, extension: ZIP}
```
also art variable is not required

trapacska commented on appetize/until.go:
```diff
@@ -0,0 +1,140 @@
+			return a.Path(), nil
+		} else if a.Ext() == APP {
+			log.Warnf("🚨 Provided file is %s", a.Ext())
+			log.Printf("Need to compress it...")
```
Wdyt about `Compressing it...`?
Because  Need to compress it doesn't means we are doing it already ;)

BirmacherAkos replied:
discussed

trapacska commented on appetize/until.go:
```diff
@@ -0,0 +1,140 @@
+}
+
+// Path returns the artifact's path
+func (a Artifact) Path() string {
```
these functions are not required as they returns only a struct field, which can be accessed from the struct itself.
(Artifact{}).Path() == (Artifact{}).filePath

BirmacherAkos replied:
discussed

trapacska commented on appetize/until.go:
```diff
@@ -0,0 +1,140 @@
+
+func ensureZIP(sourcePath string, destination string) error {
+	fmt.Println()
+	log.Infof("Creating %s from %s", destination, sourcePath)
```
we usually have logging other than in func main() that is in the same file, main.go. It is because if you want to use this package somewhere else then you need to make sure that it is logging in a handled way. (Either you can use our logger as well)

BirmacherAkos replied:
discussed

trapacska commented on step.yml:
```diff
@@ -0,0 +1,71 @@
+        The token provided by Appetize.io
+        
+        **NOTE:** You can reaquest an api token for Appetioze.io here:
+        [Appetioze.io/request-api-token](https://appetize.io/docs#request-api-token)
```
typo

BirmacherAkos committed:
PR clean  - fix

trapacska reviewed (approved):
🎅🏽
